app/importers/rows/student_section_assignment_row.rb

```diff
@@ -14,19 +14,34 @@ def self.build(row)
   end
 
   def build
-    if student and section
-      student_section_assignment = StudentSectionAssignment.find_or_initialize_by(student: student,
-                                                                                section: section)
-      return student_section_assignment
-    end
+    return if course.nil? || student.nil? || section.nil?
```

Comment:
Can we make the flow more explicit?  If I read it right, `build` executes eight database queries for each row of data, and this isn't super visible right now.  The logging also throws away the information about why theses rows can't be imported (eg, there's no student that matches, there's no course that matches, there's no section that matches).

Reply:
Sure, I can make those changes so that the importer logs more information about why a row can't be imported.

Reply:
👍 I don't know what the baseline is here (whether we're importing 100% of the rows or 63%) so might be good to see that while we're doing this, but that might be creeping too so whatever you think is best :)

Reply:
Hmmm, yes, perhaps this would be creep since our immediate goal is to fix the bad data. I'm also not too worried about 8 database queries from a performance point of view since this runs in the background.

Reply:
👍 Yeah I don't care about database queries here either, more to learn which query failed to see why the row is invalid.

Reply:
@kevinrobinson Oooh, gotcha!